python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/shape_lib_gen.py

```diff
@@ -946,6 +946,12 @@ def hacky_get_unknown_dimension_size():
 def aten〇bincount(self: List[int], weights: Optional[List[int]] = None, minlength: int = 0) -> List[int]:
     return [hacky_get_unknown_dimension_size()]
 
+def aten〇linalg_vector_norm(self: List[int], ord: float = 2, dim: Optional[List[int]] = None, keepdim: bool = False, dtype: Optional[int] = None) -> List[int]:
+    if dim is not None:
+        return upstream_shape_helpers.mean_dim(self, dim, keepdim, dtype)
+    dim_arg = list(range(len(self)))
+    return upstream_shape_helpers.mean_dim(self, dim_arg, keepdim, dtype)
```

Comment:
nit: you can do

```python
if dim is None:
  dim = list(range(len(self)))
return upstream_shape_helpers.mean_dim(self, dim, keepdim, dtype)

```

Reply:
Hmm.. I thought when I had tried it a few days ago, it complained about mismatched types for `dim`, but I don't see that error with your snippet.  Fixed it in commit 9cf993cf.  Thanks for the simpler, shorter code!